modules/sdk/project-templates/project-templates/src/main/java/com/liferay/project/templates/internal/util/JakartaCompatabilityUtil.java

```diff
@@ -114,7 +149,29 @@ private static void _updateGradleDependencies(Path gradleFilePath)
 	private static final String _TAGLIB_URL_OLD =
 		"http://java.sun.com/jsp/jstl/core";
 
+	private static final String _TLD_TAGLIB_TAG_NEW =
+		"<taglib\n\tversion=\"3.0\"\n\txmlns=" +
+			"\"https://jakarta.ee/xml/ns/jakartaee\"\n\txmlns:xsi=" +
+				"\"http://www.w3.org/2001/XMLSchema-instance\"\n\txsi:" +
+					"schemaLocation=\"https://jakarta.ee/xml/ns/jakartaee " +
+						"https://jakarta.ee/xml/ns/jakartaee" +
+							"/web-jsptaglibrary_3_0.xsd\"\n>";
+
+	private static final String _XML_WEBAPP_TAG_NEW =
+		"<web-app version=\"6.0\" " +
+			"xmlns=\"https://jakarta.ee/xml/ns/jakartaee\" " +
+				"xmlns:xsi=\"http://www.w3.org/2001/XMLSchema-instance\" " +
+					"xsi:schemaLocation=\"" +
+						"https://jakarta.ee/xml/ns/jakartaee " +
+							"https://jakarta.ee/xml/ns/jakartaee" +
+								"/web-app_6_0.xsd\">";
+
 	private static final Properties _jakartaDependenciesProperties;
+	private static final Pattern _tldTaglibTagPattern = Pattern.compile(
+		"<taglib\\n.*\\n\\t" +
+			"xmlns=\"http://java.sun.com/xml/ns/javaee\"\\n.*\\n.*\\n>");
+	private static final Pattern _xmlWebappTagPattern = Pattern.compile(
+		"^<web-app.*>$", Pattern.MULTILINE);
```

Comment:
Is there a chance of this matching too greedily? Only asking hypothetically, I can't think of a real example.

Reply:
Ah, meant to update this one to a more specific regex after initial testing. Thanks for catching!

It technically wouldn't match anything other than what we need in the project templates, but for consistency I updated it.